### DISCUSSION TITLE
Write data without internet

### DISCUSSION THREAD

**Discussion by musiolmarco** [xmartlabs/stock#24] [Q&A] [answered]
Hey,
I just found this nice blog entry (https://blog.xmartlabs.com/blog/get-flutter-offline-support-in-supabase/) and I wanted to thank you in advance for the package and I'm glad to see that you make something like this available open source. Unfortunately I don't really have much time right now and I didn't see at first glance if the package has what I need. 

I'm using Supabase as backend for a Flutter app and I've seen that you can store the data offline with the package. But is it also possible to create data anyway when no internet is available and will it be automatically synced to Supabase when internet is available again?

Or is the package only to store the data offline to make them available if there is no internet or to improve the loading times?

**Comment by f7deleon** [comment, marked as answer]
Hey Marco, 

Thanks for checking out the package.

Unfortunately, the feature that you mention is not in stock and we are not planning to include it since is not related to stock purposes.
We have worked on an application that requires this kind of functionality and usually depends on the backend communication and sync strategies. I don´t think there is a library that does this. As an alternative, you should consider using a local database and synchronize the data by using [WorkManager](https://pub.dev/packages/workmanager).

**Comment by musiolmarco** [comment]
Oh okay, thanks anyway for the answer and for your help!

I will have a look at the WorkManager.
